reckless_lib/src/utils.rs

```diff
@@ -19,17 +20,37 @@ pub fn create_dir_in_home(relative_path: &str) {
 }
 
 pub fn get_dir_path_from_url(url: &str) -> String {
-    let mut path = env::home_dir()
+    let path = env::home_dir()
         .unwrap()
         .into_os_string()
         .into_string()
         .unwrap();
-    return path + "/.reckless/" + &url.split(".com/").last().unwrap().trim().to_string();
+    return format!(
+        "{}/.reckless{}",
+        path,
+        &url.split(".com").last().unwrap().trim().to_string()
+    );
+}
+
+pub fn clone_recursive_fix(repo: git2::Repository, url: &str) {
```

Comment:
```suggestion
pub fn clone_recursive_fix(repo: git2::Repository, url: &str) -> Result<(), RecklessError> {
```

Reply:
Yes, that would have been my general idea. I will keep this is mind.